From 82ceb3f9a089104f353b9532bd6b28c82c3b65ca Mon Sep 17 00:00:00 2001
From: Kristian Amlie <kristian.amlie@mender.io>
Date: Tue, 3 May 2016 10:23:30 +0200
Subject: [PATCH] Revert "Fix memory leak with not freeing OpenSSL errors"

This reverts commit a63d0f55da87fc620c66f1b909d752d7c8b1159c.

Both Curl and OpenSSL want to call this function, but it's only
allowed to call it once. This is a bug in PHP itself, so remove for
now. A memory leak in the cleanup is insignificant compared to a
crash.
---
 ext/openssl/openssl.c |    4 ----
 1 file changed, 4 deletions(-)

diff --git a/ext/openssl/openssl.c b/ext/openssl/openssl.c
index 88e396c..27d9c6c 100644
--- a/ext/openssl/openssl.c
+++ b/ext/openssl/openssl.c
@@ -1277,10 +1277,6 @@ PHP_MSHUTDOWN_FUNCTION(openssl)
 {
 	EVP_cleanup();
 
-#if OPENSSL_VERSION_NUMBER >= 0x00090805f
-	ERR_free_strings();
-#endif
-
 	php_unregister_url_stream_wrapper("https" TSRMLS_CC);
 	php_unregister_url_stream_wrapper("ftps" TSRMLS_CC);
 
-- 
1.7.9.5

